test/integration/play/test_play.py

```diff
@@ -37,100 +29,109 @@ def test_play_non_existent_uri(tt_cmd, tmp_path):
     assert re.search(r"no connection to the host", output)
 
 
-def test_play_non_existent_file(tt_cmd, tmp_path, test_instance):
-    # Run play with non-existent file.
-    cmd = [tt_cmd, "play", "127.0.0.1:" + test_instance.port, "path-to-non-existent-file"]
-    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
-    assert rc == 1
-    assert re.search(r"No such file or directory", output)
+# -----------------------------------------------------------------------------
+# Failed tests
+def make_test_play_param_failed(
+    args={},
+    play_error="",
+):
+    return pytest.param(args, play_error)
 
 
-def test_play_test_remote_instance(tt_cmd, test_instance):
+@pytest.mark.parametrize("args, play_error", [
+    make_test_play_param_failed(
+        # Testing with unset uri and .xlog or .snap file.
+        play_error="required to specify an URI and at least one .xlog or .snap file",
+    ),
+    make_test_play_param_failed(
+        args={"path-to-non-existent-file"},
+        play_error="No such file or directory",
+    ),
+    make_test_play_param_failed(
+        args={"test.xlog", "--timestamp=abcdef", "--space=999"},
+        play_error='failed to parse a timestamp: parsing time "abcdef"',
+    ),
+    make_test_play_param_failed(
+        args={"test.xlog", "--timestamp=2024-11-14T14:02:36.abc", "--space=999"},
+        play_error='failed to parse a timestamp: parsing time "2024-11-14T14:02:36.abc"',
+    ),
+])
+def test_play_test_remote_instance_timestamp(tt_cmd, test_instance, args, play_error):
     # Play .xlog file to the remote instance.
-    cmd = [tt_cmd, "play", "127.0.0.1:" + test_instance.port, "test.xlog", "--space=999"]
-    rc, output = run_command_and_get_output(cmd, cwd=test_instance._tmpdir)
-    assert rc == 0
-    assert re.search(r"Play result: completed successfully", output)
-
-    # Testing played .xlog file from the remote instance.
-    cmd = [tt_cmd, "cat", "00000000000000000000.xlog", "--space=999"]
-    rc, output = run_command_and_get_output(cmd, cwd=test_instance._tmpdir)
-    assert rc == 0
-    assert re.search(r"space_id: 999", output)
-    assert re.search(r"[1, 'Roxette', 1986]", output)
-    assert re.search(r"[2, 'Scorpions', 2015]", output)
-    assert re.search(r"[3, 'Ace of Base', 1993]", output)
-
-
-TEST_PLAY_TIMESTAMP_PARAMS_CCONFIG = ("input, play_result, found, not_found")
+    cmd = [tt_cmd, "play", "127.0.0.1:" + test_instance.port]
```

Comment:
```suggestion
    cmd = [tt_cmd, "play", f"127.0.0.1:{test_instance.port}"]
```
What happens with `add` in case if `port` would become an integer?

Reply:
[It looks impossible.](https://github.com/tarantool/tt/blob/7d8a6cfc027e2b9403c1a77e0b35c32cf25feaed/test/utils.py#L284)

Reply:
Not exactly, there is comment info, only.
Suggested solution works well with any type of variable, while "add" only with strings.